scripts/demangle.py

```diff
@@ -59,46 +75,123 @@ def demangle(func: str):
     #     # Demangling failed
     #     print("Demangling failed with status", status)
 
-for json_file_name in file_names:
-    file_name = json_file_name.split('.json')[0]
-    # Find file path
-    if len(glob.glob(proj_path+'/'+file_name+'.c*'))==0:
-        print('ERROR: File '+proj_path+'/'+file_name+'.c*'+" NOT found")
-        continue
-    file_path = glob.glob(proj_path+'/'+file_name+'.c*')[0]
-    # if file_name == 'augeas_api_fuzzer': continue
-    # open Json file and get filename + func name
-    with open(json_path+'/'+json_file_name, 'r') as f:
-        try:
-            data = json.load(f)
-        except:
+
+# get source code from docker
+def get_source_code(file_path: str, file_name: str, proj_name: str):
+    # Create a container from the image
+    client = docker.from_env()
+    image = client.images.get("gcr.io/oss-fuzz/" + proj_name + ":latest")
+    container = client.containers.run(image, detach=True)
+    # check file_path, it may not the absolut path
+    if not "/" in file_path:
+        file_path = "/src/" + proj_name + "/" + file_path
+    try:
+        code_content = ""
+        for chunk in container.get_archive(file_path)[0]:
+            code_content += chunk.decode("utf-8")
+    except docker.errors.NotFound:
+        code_content = "ERROR"
+        warning(f"ERROR: File  {file_path}  NOT found\n")
+    # Stop and remove the container
+    container.stop()
+    container.remove()
+    return code_content
+
+
+def main():
+    parser = argparse.ArgumentParser(description="Catch source code")
+    parser.add_argument(
+        "-n",
+        "--name",
+        type=str,
+        required=False,
+        default="coturn",
+        help="The project name to fetch",
+    )
+    args = parser.parse_args()
+    proj_name = args.name
+    OSSFUZZ_SCRIPTS_HOME = ".."
+    json_path = os.path.join(OSSFUZZ_SCRIPTS_HOME, "dump", proj_name)
+    json_file_names = [f for f in os.listdir(json_path) if f.endswith(".json")]
+    info(f"Looking for json file: {json_file_names}")
+
+    for json_file_name in json_file_names:
+        file_name = json_file_name.split(".json")[0]
+        # open Json file and get filename + func_name
+        with open(json_path + "/" + json_file_name, "r") as f:
+            try:
+                data = json.load(f)
+            except json.JSONDecodeError:
+                try:
+                    # the json file start with path, reopen the json file
+                    with open(json_path + "/" + json_file_name, "r") as f:
```

Comment:
In what case does json file starts with path?

Reply:
In some JSON files, it starts with metadata and then followed by JSON data, thus I need to remove those metadata and read json.

e.g.
`rm: cannot remove '/tmp/basicstuff_fuzzer_corpus': Device or resource busy
/out/basicstuff_fuzzer -rss_limit_mb=2560 -timeout=25 \-max_total_time=60 \-rss_limit_mb=0 /tmp/basicstuff_fuzzer_corpus < /dev/null
INFO: Running with entropic power schedule (0xFF, 100).
INFO: Seed: 760076011
INFO: Loaded 1 modules   (5109 inline 8-bit counters): 5109 [0x7e11b0, 0x7e25a5), 
{
			"/src/basicstuff_fuzzer__GLOBAL__sub_I_basicstuff_fuzzer.cc": [
				[
					[
						[
							"0"
						],
						[]
					]
				]
			]
}`

Reply:
`rm: cannot remove...` is message to stderr, which shouldn't exists in the newest version of script since its not reading from stderr. Could you verify this in the dataset @harrison4anewbe? thanks

Reply:
Yes, the latest version of JSON should have no stderr, I have removed this redundant code.